src/sinks/datadog/logs.rs

```diff
@@ -2,59 +2,116 @@ use crate::{
     config::{log_schema, DataType, SinkConfig, SinkContext, SinkDescription},
     event::Event,
     sinks::{
-        util::{self, encoding::EncodingConfig, tcp::TcpSink, Encoding, StreamSinkOld, UriSerde},
+        util::{
+            encode_event,
+            encoding::{EncodingConfig, EncodingConfiguration},
+            http::{BatchedHttpSink, HttpClient, HttpSink},
+            BatchConfig, BatchSettings, BoxedRawValue, Encoding, JsonArrayBuffer,
+            TowerRequestConfig, VecBuffer,
+        },
         Healthcheck, VectorSink,
     },
-    tls::{MaybeTlsSettings, TlsConfig},
 };
 use bytes::Bytes;
-use futures01::{stream::iter_ok, Sink};
+use futures::FutureExt;
+use futures01::Sink;
+use http::{Request, StatusCode};
+use hyper::body::Body;
 use serde::{Deserialize, Serialize};
+use serde_json::json;
 
-#[derive(Deserialize, Serialize, Debug)]
+#[derive(Deserialize, Serialize, Debug, Clone)]
 #[serde(deny_unknown_fields)]
 pub struct DatadogLogsConfig {
-    endpoint: Option<UriSerde>,
+    endpoint: Option<String>,
     api_key: String,
     encoding: EncodingConfig<Encoding>,
-    tls: Option<TlsConfig>,
+
+    #[serde(default)]
+    batch: BatchConfig,
+
+    #[serde(default)]
+    request: TowerRequestConfig,
+}
+
+#[derive(Clone)]
+pub struct DatadogLogsJsonService {
+    config: DatadogLogsConfig,
+}
+
+#[derive(Clone)]
+pub struct DatadogLogsTextService {
+    config: DatadogLogsConfig,
 }
 
 inventory::submit! {
     SinkDescription::new_without_default::<DatadogLogsConfig>("datadog_logs")
 }
 
+impl DatadogLogsConfig {
+    fn get_endpoint(&self) -> &str {
+        match &self.endpoint {
+            Some(ref endpoint) => &endpoint,
+            None => "https://http-intake.logs.datadoghq.eu/v1/input",
+        }
```

Comment:
Is not too much refs here? Should work:
```suggestion
        match &self.endpoint {
            Some(endpoint) => endpoint,
            None => "https://http-intake.logs.datadoghq.eu/v1/input",
        }
```

Or as alternative, can be changed to:
```rust
self.endpoint.as_ref().map(|s| s.as_str()).unwrap_or("https://http-intake.logs.datadoghq.eu/v1/input")
```